$OpenBSD: patch-libraries_liblutil_uuid_c,v 1.1 2020/06/18 21:36:11 sthen Exp $

OpenLDAP's first choice uuid functions are close to but not quite the same
as we have in libc, differences being that we don't have uuid_to_str() (in
fact I don't see which other OS might have it either, no other software in
debian codesearch uses this), and it needs uuid.h not sys/uuid.h.

Patch to use it, avoiding the need for Linux libuuid.

Index: libraries/liblutil/uuid.c
--- libraries/liblutil/uuid.c.orig
+++ libraries/liblutil/uuid.c
@@ -40,7 +40,7 @@
 #include <ac/string.h>	/* get memcmp() */
 
 #ifdef HAVE_UUID_TO_STR
-#  include <sys/uuid.h>
+#  include <uuid.h>
 #elif defined( HAVE_UUID_GENERATE )
 #  include <uuid/uuid.h>
 #elif defined( _WIN32 )
@@ -284,7 +284,7 @@ lutil_uuidstr( char *buf, size_t len )
 		return 0;
 	}
 
-	uuid_to_str( &uu, &s, &rc );
+	uuid_to_string( &uu, &s, &rc );
 	if ( rc != uuid_s_ok ) {
 		return 0;
 	}
